### PR TITLE
Feat/authenticate action

### PR DESCRIPTION
**Pull request by djvs**

### CODE REVIEW

**Comment by djvs** [comment]
Provides additional action type to support attestation-less authentication via signature.

File: src/types.ts [hunk 1]
```diff
@@ -5,19 +5,31 @@ import Bowser from 'bowser'
 
 enum Action {
   attestation = 'request_attestation_data',
+  authentication = 'authentication',
 }
 
-type RequestData = {
-  action: Action
+interface IRequestDataBase {
+  action: keyof typeof Action
```
Comment (JaceHensley) [inline src/types.ts]: This is a breaking change, and will cause the mobile apps to break

File: src/types.ts [hunk 2]
```diff
@@ -5,19 +5,31 @@ import Bowser from 'bowser'
 
 enum Action {
   attestation = 'request_attestation_data',
+  authentication = 'authentication',
 }
 
-type RequestData = {
-  action: Action
+interface IRequestDataBase {
```
Comment (JaceHensley) [inline src/types.ts]: Let's just keep these as `type`s for consistency purposes

File: src/types.ts [hunk 3]
```diff
@@ -5,19 +5,31 @@ import Bowser from 'bowser'
...
   token: string
   url: string
   org_logo_url: string
   org_name: string
   org_usage_policy_url: string
   org_privacy_policy_url: string
+}
+
+type RequestDataAttestation = RequestDataBase & {
+  action: 'attestation'
```
Comment (JaceHensley) [inline src/types.ts]: I think this would need to be `'request_attestation_data'`

**Pull request merged**